tuf/data/roles.go

```diff
@@ -274,7 +272,9 @@ func mergeStrSlices(orig, new []string) []string {
 	}
 	for _, e := range new {
 		if !have[e] {
-			orig = append(orig, e)
+			origCopy := make([]string, len(orig))
+			copy(origCopy, orig)
+			orig = append(origCopy, e)
```

Comment:
I'm not sure if this is the most go-esque way to do what you had in mind @aaronlehmann (using a copy to avoid slice corruption), please let me know if you think there's a better way to do this!